src/sql/src/normalize.rs

```diff
@@ -306,6 +306,14 @@ pub fn create_statement(
             *name = allocate_name(name)?;
             *if_not_exists = false;
             *materialized = false;
+
+            for opt in with_options.iter_mut() {
+                if let SqlOption::ObjectName { name, object_name } = opt {
+                    if ident_ref(name) == "tx_metadata" {
+                        *object_name = allocate_name(object_name)?;
```

Comment:
So I think `allocate_name` will probably result in the right name most of the time, but I don't think it's exactly correct. Since `object_name` is an existing object, we should probably be looking for it in the catalog instead of assuming it's database and schema which `allocate_name` sort of does.

Instead you can resolve the object with with `scx.resolve_item` and get the fully qualified name that way.

Another option, which I think is probably better but relies on `object_name` always being a valid catalog object, is to apply the following patch (I don't know how to attach a patch on GH):
```
From 95b3de8247aff77ec9e8af728234c26b76e6a561 Mon Sep 17 00:00:00 2001
From: Joseph Koshakow <koshy44@gmail.com>
Date: Wed, 30 Mar 2022 17:23:06 -0400
Subject: [PATCH] Auto resolve

---
 src/sql-parser/src/ast/defs/statement.rs | 2 +-
 src/sql-parser/src/parser.rs             | 2 +-
 src/sql/src/normalize.rs                 | 8 --------
 3 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/src/sql-parser/src/ast/defs/statement.rs b/src/sql-parser/src/ast/defs/statement.rs
index cbe666193..d56d0c684 100644
--- a/src/sql-parser/src/ast/defs/statement.rs
+++ b/src/sql-parser/src/ast/defs/statement.rs
@@ -1698,7 +1698,7 @@ pub enum SqlOption<T: AstInfo> {
     },
     ObjectName {
         name: Ident,
-        object_name: UnresolvedObjectName,
+        object_name: T::ObjectName,
     },
     DataType {
         name: Ident,
diff --git a/src/sql-parser/src/parser.rs b/src/sql-parser/src/parser.rs
index 6351aee1c..66cf272e7 100644
--- a/src/sql-parser/src/parser.rs
+++ b/src/sql-parser/src/parser.rs
@@ -2849,7 +2849,7 @@ impl<'a> Parser<'a> {
             SqlOption::Value { name, value }
         } else {
             self.prev_token();
-            if let Ok(object_name) = self.parse_object_name() {
+            if let Ok(object_name) = self.parse_raw_name() {
                 SqlOption::ObjectName { name, object_name }
             } else {
                 self.expected(self.peek_pos(), "option value", token)?
diff --git a/src/sql/src/normalize.rs b/src/sql/src/normalize.rs
index 4df95066f..cc2eda18b 100644
--- a/src/sql/src/normalize.rs
+++ b/src/sql/src/normalize.rs
@@ -306,14 +306,6 @@ pub fn create_statement(
             *name = allocate_name(name)?;
             *if_not_exists = false;
             *materialized = false;
-
-            for opt in with_options.iter_mut() {
-                if let SqlOption::ObjectName { name, object_name } = opt {
-                    if ident_ref(name) == "tx_metadata" {
-                        *object_name = allocate_name(object_name)?;
-                    }
-                }
-            }
         }
 
         Statement::CreateTable(CreateTableStatement {
```

That will automatically resolve and full qualify `object_name` as part of planning.

Reply:
Thanks for writing up the patch! I'm not sure if that generalizes / I don't want to wrap that change into this PR but it could be good to look into for the future -- and I do like the idea of not special casing tx_metadata but alas

I'll make a change to take the `resolve_item` route as you suggest